SAMLPortal/Controllers/SetupController.cs

```diff
@@ -0,0 +1,379 @@
+			}
+
+			var ldapSearchBase = GlobalSettings.Get("LDAP_SearchBase");
+			var ldapAdminFilter = GlobalSettings.Get("LDAP_AdminFilter");
+			var ldapUserFilter = GlobalSettings.Get("LDAP_UsersFilter");
+
+			StepFourModel model = new StepFourModel();
+			model.SearchBase = ldapSearchBase != null ? ldapSearchBase : "";
+			model.AdminFilter = ldapAdminFilter != null ? ldapAdminFilter : "uid={0}";
+			model.UserFilter = ldapUserFilter != null ? ldapUserFilter : "uid={0}";
+
+			return View(model);
+		}
+
+		[HttpPost]
+		[Route("4")]
+		public IActionResult StepFour(StepFourModel model)
+		{
+			if (!CheckStep(4))
+			{
+				return Redirect("3");
+			}
+
+			if (ModelState.IsValid && model != null)
+			{
+				GlobalSettings.Store("LDAP_SearchBase", model.SearchBase);
+				GlobalSettings.Store("LDAP_AdminFilter", model.AdminFilter);
+				GlobalSettings.Store("LDAP_UsersFilter", model.UserFilter);
+
+				Helpers.ReplaceEnvVariableInFile(GlobalSettings.Get("CONFIG_FILE"), "SP_CONFIG_SETUPASSISTANT_STEP", "5");
+				GlobalSettings.Store("CONFIG_SETUPASSISTANT_STEP", "5");
+			}
+
+			return View(model);
+		}
+
+		[HttpGet]
+		[Route("5")]
+		public IActionResult StepFive()
+		{
+			if (!CheckStep(5))
+			{
+				return Redirect("4");
+			}
+
+			var ldapDisplayName = GlobalSettings.Get("LDAP_Attr_DisplayName");
+			var ldapUID = GlobalSettings.Get("LDAP_Attr_UID");
+			var ldapMemberOf = GlobalSettings.Get("LDAP_Attr_MemberOf");
+			var ldapMail = GlobalSettings.Get("LDAP_Attr_Mail");
+
+			StepFiveModel model = new StepFiveModel();
+			model.DisplayName = ldapDisplayName != null ? ldapDisplayName : "cn";
+			model.UID = ldapUID != null ? ldapUID : "uid";
+			model.MemberOf = ldapMemberOf != null ? ldapMemberOf : "memberOf";
+			model.Mail = ldapMail != null ? ldapMail : "mail";
+
+			return View(model);
+		}
+
+		[HttpPost]
+		[Route("5")]
+		public IActionResult StepFive(StepFiveModel model)
+		{
+			if (!CheckStep(5))
+			{
+				return Redirect("4");
+			}
+
+			if (ModelState.IsValid && model != null)
+			{
+				GlobalSettings.Store("LDAP_Attr_DisplayName", model.DisplayName);
+				GlobalSettings.Store("LDAP_Attr_UID", model.UID);
+				GlobalSettings.Store("LDAP_Attr_MemberOf", model.MemberOf);
+				GlobalSettings.Store("LDAP_Attr_Mail", model.Mail);
+
+				Helpers.ReplaceEnvVariableInFile(GlobalSettings.Get("CONFIG_FILE"), "SP_CONFIG_SETUPASSISTANT_STEP", "6");
+				GlobalSettings.Store("CONFIG_SETUPASSISTANT_STEP", "6");
+
+				return Redirect("/");
+			}
+
+			return View(model);
+		}
+
+		/// <summary>
+		/// Returns false if the user purposely miss a step
+		/// </summary>
+		/// <param name="step"></param>
+		/// <returns></returns>
+		private static bool CheckStep(int step)
+		{
+			if (step - GlobalSettings.GetInt("CONFIG_SETUPASSISTANT_STEP") <= 1)
+			{
+				return true;
+			}
+			else
+			{
+				return false;
+			}
+		}
+	}
+
+	public class SetupAsyncActionFilter : IAsyncActionFilter
+	{
+		ActionExecutionDelegate _next;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this unread private field '_next' or refactor the code to use its value.](https://app.codacy.com/manual/zegorax/SAMLPortal/pullRequest?prid=5343548)